src/ppx/application.cpp

```diff
@@ -937,11 +942,29 @@ int Application::Run(int argc, char** argv)
     }
     mStandardOptions = mCommandLineParser.GetOptions().GetStandardOptions();
 
+    // Knobs need to be set up before commandline parsing.
+    DispatchKnobs();
+
+    // Append Knob flags to usage message
+    if (!mKnobManager.IsEmpty()) {
+        mCommandLineParser.AppendUsageMsg(mKnobManager.GetUsageMsg());
+    }
+
     if (mStandardOptions.help) {
         PPX_LOG_INFO(mCommandLineParser.GetUsageMsg());
         return EXIT_SUCCESS;
     }
 
+    // Parse knobs
+    if (!mKnobManager.IsEmpty()) {
+        auto options = mCommandLineParser.GetOptions();
+        auto res     = mKnobManager.UpdateFromFlags(options);
```

Comment:
Not sure this is needed.
You could definitely build knobs using the flag values directly, so this whole API surface can be removed.